arcgis-ios-sdk-samples/Extensions/UIKit/UIViewController.swift

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+extension UIViewController {
+    
+    func presentAlert(title: String? = nil, message: String) {
```

Comment:
For what it's worth, based on my experience, it is better to keep view controller creation and presentation separate. One reason is testing. If the alert controller was created through an initializer, it would be easy to write an `testInitWithBlah()` function, create one, and assert that it was initialized properly. By conflating creation with presentation, that test becomes much more complicated. It has to create a another view controller to do the presenting. It has to wait for the animation to finish (assuming it knows the presentation will be animated--more on that in a bit). It has to get the presented view controller and cast it to an alert controller. In the end, it is testing view controller presentation as much as it is testing initialization of the alert controller.

I would encourage you to write tests for this code. And speaking of documentation, please add documentation comments for both of these methods. That will really help anybody else who comes across these methods being called and wonders what they do.